File: Front_Alumno2024/src/app/model/Alumno.ts
export class Alumno {
    id: number;
    nombre: string;
    apellido: string;
    dni: string;
    ciclo: number;
    estado: string;
    usuario: string;
    fecha: Date;  
  
    constructor(
      id: number =0,
      nombre: string='',
      apellido: string='',
      dni: string='',
      ciclo: number=0,
      estado: string='',
      usuario: string='',
      fecha: Date = new Date()
    ) {
      this.id = id;
      this.nombre = nombre;
      this.apellido = apellido;
      this.dni = dni;
      this.ciclo = ciclo;
      this.estado = estado;
      this.usuario = usuario;
      this.fecha = fecha;
    }
  }
  

File: Back_Alumno2024/src/main/java/com/app/model/Alumno.java
package com.app.model;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "ALUMNO")
public class Alumno {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nombre;

    @Column(nullable = false)
    private String apellido;

    @Column(nullable = false, length = 8)
    private String dni;

    @Column(nullable = false)
    private int ciclo;

    @Column(nullable = false, length = 1)
    private String estado;

    @Column(name = "nombre_usuario")
    private String usuario;

    @Column(nullable = false)
    private LocalDateTime fecha;
}
